apps/documents/serializers.py

```diff
@@ -0,0 +1,143 @@
+from django.conf import settings
+
+from enumfields.drf import EnumField
+from rest_framework_json_api import serializers
+from rest_framework.utils.serializer_helpers import ReturnList
+
+from apps.utils import get_s3_client
+from .models import Document, DocumentType, FileType, UploadStatus
+
+
+class DocumentCreateSerializer(serializers.ModelSerializer):
+    """
+    Model serializer for documents validation for s3 signing
+    """
+    document_type = EnumField(DocumentType, ints_as_names=True)
+    file_type = EnumField(FileType, ints_as_names=True)
+    upload_status = EnumField(UploadStatus, ints_as_names=True)
+    shipment_id = serializers.CharField(max_length=36)
+
+    class Meta:
+        model = Document
+        fields = ['name', 'description', 'document_type', 'file_type', 'size', 'upload_status', 'shipment_id']
+
+    def s3_sign(self, document=None):   # pylint: disable=too-many-locals
+        data = self.validated_data
+        s3, _ = get_s3_client()     # pylint: disable=invalid-name
+        bucket = settings.S3_BUCKET
+        file_type = data['file_type']
+        file_type_name = file_type.name.lower()
+
+        # Allowed files type list
+        img_types_list = [enum.name.lower() for enum in FileType]
+        img_types_list.pop(img_types_list.index('pdf'))
+
+        if file_type_name in img_types_list:
+            content_type = f"image/{file_type_name}"
+        else:
+            content_type = f"application/{file_type_name}"
+
+        shipment = document.shipment
+        wallet_id = shipment.shipper_wallet_id
+        storage_credentials_id = shipment.storage_credentials_id
+        vault_id = shipment.vault_id
```

Comment:
I don't think we need these 4 assignments, they just take up space

Reply:
This is to avoid a statement longer than 120 characters which will trigger pylint error

Reply:
that looks great!